src/java.base/share/classes/java/net/InetAddress.java

```diff
@@ -797,9 +800,10 @@ String getHostName(boolean check) {
      * If the operation is not allowed, it will return
      * the textual representation of the IP address.
      *
-     * @return  the fully qualified domain name for this IP address,
-     *    or if the operation is not allowed by the security check,
-     *    the textual representation of the IP address.
+     * @return  the fully qualified domain name for this IP address, or the textual representation
+     *          of the IP address if either the operation is not allowed by the security check
+     *          or the system-wide resolver wasn't able to determine the fully qualified domain
+     *          name for the IP address.
```

Comment:
Re-reading this, I wonder if a better wording would be:

```suggestion
     * @return  the fully qualified domain name for this IP address. 
     *                If either the operation is not allowed by the security check
     *                or the system-wide resolver wasn't able to determine the fully qualified domain
     *                name for the IP address, the textual representation of the IP address is returned instead.
```

Just a suggestion.

Reply:
The suggested change sounds fine to me.